Sources/AWSSDKSwiftCore/AWSShapes/Payload.swift

```diff
@@ -16,44 +16,106 @@ import struct Foundation.Data
 import NIO
 import NIOFoundationCompat
 
-/// Object storing request/response payload
+/// Holds a request or response payload. A request payload can be in the form of either a ByteBuffer or a stream function that will supply ByteBuffers to the HTTP client.
+/// A response payload only comes in the form of a ByteBuffer
 public struct AWSPayload {
-
-    /// Construct a payload from a ByteBuffer
-    public static func byteBuffer(_ byteBuffer: ByteBuffer) -> Self {
-        return Self(byteBuffer: byteBuffer)
+    
+    /// Internal enum
+    enum Payload {
+        case byteBuffer(ByteBuffer)
+        case stream(size: Int?, stream: (EventLoop)->EventLoopFuture<ByteBuffer>)
```

Comment:
move size to the back and make an `UInt?`

Reply:
what do you think about adding a `.none` case? that way we don't need to make `AWSPayload` an optional everywhere.

Reply:
Given most people are going to define a stream as follows
```
let payload = .stream(size:16384) {
   ...
}
```
I'm not sure the order matters
Also no point setting the size to `UInt` when the size in AHC is an `Int`

Reply:
`Int.max` would support a stream size of 9 exabytes.

Reply:
Regarding the `.none` payload (I prefer `.empty`). Yeah I can look into that. I'll add an issue to look into after though.

Reply:
I'm not sure if it is a good idea to postpone `empty` until a next pr. we change the AWSPayload in this pr anyway. I'm not sure, if we want two half changes?

Your call... Just raising the issue!

Reply:
I can do, just trying to keep PR smallish. I guess this isnt a huge addition

Reply:
done